std/experimental/allocator/mallocator.d

```diff
@@ -210,17 +225,28 @@ struct AlignedMallocator
     $(WEB msdn.microsoft.com/en-us/library/8z34s9c6(v=vs.80).aspx,
     $(D __aligned_malloc)) on Windows.
     */
-    version(Posix) @trusted @nogc
+    version(Posix)
+    @trusted @nogc nothrow
     void[] alignedAllocate(size_t bytes, uint a) shared
     {
-        import core.stdc.errno : ENOMEM;
+        import core.stdc.errno : ENOMEM, EINVAL;
         assert(a.isGoodDynamicAlignment);
         void* result;
         auto code = posix_memalign(&result, a, bytes);
-        if (code == ENOMEM) return null;
-        return result[0 .. bytes];
+        if (code == ENOMEM)
+            return null;
+
+        else if (code == EINVAL)
+            assert (0, "AlignedMallocator.alignment is not a power of two multiple of (void*).sizeof, according to posix_memalign!");
```

Comment:
Yes, it behaves exactly the same, just a HLT. On Posix systems (which obviously this will be on), this manifests as a segmentation fault. This is why I created abort as mentioned above -- no point in having a message generated when it is ignored.